### PR TITLE
docs: optimize build

### PR DESCRIPTION
**Pull request by userquin**
## Description

This PR includes:
- ~~add `type: module` to the docs package.json file~~
- use `moduleResolution: Bundler` in tsconfig.json file
- update `@mdi/js` and `@mdi/svg` to latest version 7.4.47
- updates `plugins/icons.ts` module to include missing `mdiDiscord `, `mdiFormatTextdirectionLToR`  and `mdiFormatTextdirectionRToL` svg icons from `v6` version and removes `mdiCreationOutline` included in 7.4.47
- include a vite virtual module for mdi js names and aliases from `@mdi/svg/meta.json`
- added [mdi-js-es](https://github.com/userquin/mdi-js-es) to allow tree shaking mdi icons used for mdi icon set (`plugins/icons.ts` module): now vuetify plugin will include only mdi icons used and not the full barrel
- update Vite to `5.4.0-beta.1`  to allow use new sass modern api and enable `css.preprocessorMaxWorkers`: this will reduce ~4s starting dev server
- updates `components/docs/IconList.vue` sfc:
  - use dynamic import to load `@mdi/svg` and `@mdi/js`: this change reduces ~3MB  the index.js entry point, creating 2 new js assets that are loaded only on `features/icon-fonts` page: `@mdi/svg` moved to vite plugin
  - update logic to filter icons using iterators
  - fix small bug when including spaces and deleting them: the browser hangs
  - add clearable and loading attributes to the autocomplete: using primary color for loading ~~using color from the autocomplete, ping me if you want loading progress with primary color~~

TODO: Maybe we need to use `overrides` or `pnpm.overrides` in docs package.json and dedupe dependencies.

From this:

imagen

to this:

imagen

## Markup:

NA

### CODE REVIEW

**Comment by userquin** [comment]
Exporting only `@mdi/svg/meta.json` name and aliases using vite virtual module, we remove 2MB:

image

**Comment by userquin** [comment]
Tmr I'll remove `@mdi/js` from icons and create a new icon set using unocss preset icons: `unocss-mdi`.

This way we can remove `@mdi/js` from the index assets and all mdi icons will go to css: once done, I'll change `components/docs/IconList.vue` to load `@mdi/js` also with dynamic import.

File: packages/docs/vite.config.mts [hunk 1]
```diff
@@ -236,6 +236,27 @@ export default defineConfig(({ command, mode, isSsrBuild }) => {
...
+        name: 'vuetify:mdi-js-icons',
+        enforce: 'pre',
+        resolveId(id) {
+          if (id === 'virtual:mdi-js-icons') {
+            return `\0virtual:mdi-js-icons`
+          }
+        },
+        async load (id) {
+          if (id === `\0virtual:mdi-js-icons`) {
+            const icons: { name: string, aliases: string[] }[] = await import('@mdi/svg/meta.json', {
```
Comment (userquin) [inline packages/docs/vite.config.mts:249]: Use read file instead dynamic import, depending on node version this can fail

Comment (KaelWD) [reply packages/docs/vite.config.mts:249]: We already use this in api-gen and enforce node >18

**Review by johnleider — DISMISSED** [review]

**Comment by userquin** [comment]
not yet ready, applying some more optimizations in my local

**Comment by userquin** [comment]
Including `mdi-js-es` `package and using dynamic import to load `@mdi/js` and `@mdi/svg`:

imagen

**Comment by userquin** [comment]
Now it is ready @johnleider @MajesticPotatoe 

If you want to play with the Node Profiler:
- `pnpm vite dev --profile --open`
- await page loaded then back to the terminal: press `p + enter` (stops node profiler) and finally press `q + enter`
- then go to speedscope.app and upload generated vite-profile-0.cpuprofile in docs root folder

**Comment by KaelWD** [comment]
Screenshot_20240804_164617

Anyone know if we actually use still use the vue compiler at runtime anymore?

**Comment by KaelWD** [comment]
Screenshot_20240805_013419

**Comment by KaelWD** [comment]
https://github.com/unplugin/unplugin-vue-components/pull/788 Will replace the patch

https://github.com/cosmicjs/cosmic-sdk-js/pull/36 gets rid of axios too.

I think all the vuetify/one stuff has to go to one.vuetifyjs.com only, keep the login menu but management and subscribing can just be a separate page.

**Comment by userquin** [comment]
On Windows (command prompt), without the windows plugin, building the docs I get this error:
imagen

**Comment by KaelWD** [comment]
New preview, 320K index: https://vuetify-3-deihws1i1-vuetifyjs1.vercel.app/

Now (52):
Screenshot_20240808_015530


Live (34):
Screenshot_20240808_015541

**Pull request merged**